context/contextIsShowingPanelMenu.tsx

```diff
@@ -0,0 +1,48 @@
+import {
+	PropsWithChildren,
+	createContext,
+	useContext,
+	useEffect,
+	useState,
+} from 'react';
+
+type ShowingPanelContextType = {
+	isOpen: boolean;
+	openPanelMenu: () => void;
+	closePanelMenu: () => void;
+};
+const ShowingPanelMenuContext = createContext<
+	ShowingPanelContextType | undefined
+>(undefined);
+
+export function ShowingPanelMenuProvider({ children }: PropsWithChildren) {
+	const [isOpen, setIsOpen] = useState(true);
+	const changeOverflowBody = () => {
+		const bodyElement = document.body;
+		bodyElement.style.overflow = isOpen ? 'visible' : 'hidden';
+	};
+	useEffect(() => {
```

Comment:
you don't need this useEffect, put this changeOverflowBody to openPanelMenu and closePanelMenu